app/tests/profiles_tests/test_views.py

```diff
@@ -179,3 +180,74 @@ def test_organization_update(self, client):
         )
 
         assert org1.title not in response.content.decode()
+
+
+@pytest.mark.parametrize(
+    "viewname,subscription_attr",
+    (
+        ("newsletter-unsubscribe", "receive_newsletter"),
+        ("notification-unsubscribe", "receive_notification_emails"),
+    ),
+)
+@pytest.mark.django_db
+def test_one_click_unsubscribe(client, settings, viewname, subscription_attr):
```

Comment:
It's Django functionality by we should also check that modifying the token (e.g. changing the username) does not allow users to (un)subscribe other users.